cmd/ausoceantv/main.go

```diff
@@ -68,27 +94,70 @@ func main() {
 
 	var host string
 	var port int
-	flag.BoolVar(&app.debug, "debug", false, "Run in debug mode.")
-	flag.BoolVar(&app.standalone, "standalone", false, "Run in standalone mode.")
+	flag.BoolVar(&svc.debug, "debug", false, "Run in debug mode.")
+	flag.BoolVar(&svc.standalone, "standalone", false, "Run in standalone mode.")
 	flag.StringVar(&host, "host", "localhost", "Host we run on in standalone mode")
 	flag.IntVar(&port, "port", defaultPort, "Port we listen on in standalone mode")
-	flag.StringVar(&app.storePath, "filestore", "store", "File store path")
+	flag.StringVar(&svc.storePath, "filestore", "store", "File store path")
 	flag.Parse()
 
 	// Perform one-time setup or bail.
 	ctx := context.Background()
-	app.setup(ctx)
+	svc.setup(ctx)
+
+	// Create app.
+	app := fiber.New(fiber.Config{ErrorHandler: errorHandler})
+
+	// Recover from panics.
+	app.Use(recover.New())
+
+	// CORS middleware.
+	app.Use(cors.New(cors.Config{
+		AllowOrigins: "*",
+	}))
+
+	// Attach global service to request context.
+	app.Use(serviceMiddleware(svc))
+
+	// Set the logging level.
+	if svc.debug {
+		log.SetLevel(log.LevelDebug)
+	} else if svc.standalone {
+		log.SetLevel(log.LevelInfo)
+	} else {
+		// Appengine logs requests for us.
+		log.SetLevel(log.LevelError)
+	}
+
+	// Add logging middleware to log requests if applicable.
+	app.Use(func(ctx *fiber.Ctx) error {
+		log.Info(ctx.Path())
+		return ctx.Next()
+	})
 
-	http.HandleFunc("/api/", app.apiHandler)
+	// Register routes.
+	registerAPIRoutes(app)
 
-	log.Printf("Listening on %s:%d", host, port)
-	log.Fatal(http.ListenAndServe(fmt.Sprintf("%s:%d", host, port), nil))
+	// Start web server.
+	listenOn := fmt.Sprintf(":%d", port)
+	fmt.Printf("starting web server on %s\n", listenOn)
+	app.Listen(listenOn)
```

Comment:
We need to handle the error for Listen, perhaps wrap it in log.Fatal like it was before.